Automattic/wp-calypso
# Add-ons: Storage as a tiered-volume product [WIP]

Related to #

## Proposed Changes

* Allows an add-on to have a quantity, both for the checkout URL with quantity meta, and for calculating display prices.
* I will revisit this, and do some performance fixes where I can. I will also likely use the existing selectors for tiered-volume products, e.g. `getPriceTierForUnits`.
* This is not _the_ design. We will likely combine these together, and have a dropdown (so the add-on card will have state and re-render when the quantity of GB is selected. I also need to add the real icons. Mostly for testing and scoping out the needed work.

## To do
- [ ] Handle translations (when final copy is ready)
- [ ] Icons
- [ ] Performance, e.g. now the key for iteration over the add-ons list is the product slug, which now isn't unique. 

## Testing Instructions



* Confirm that existing checkout links work
* Confirm that the add-on links for 50 and 100 GB storage works.
* To see the grid, you need to patch backend code: D103440-code, since these are not in prod.

## Pre-merge Checklist
